opendse-model/src/main/java/net/sf/opendse/model/Specification.java

```diff
@@ -32,8 +32,11 @@
  * @author Martin Lukasiewycz
  * 
  */
-public class Specification implements IAttributes {
+public class Specification extends Element {
 
+	// the unique id of the specification element; no other element must have this id
+	public static final String SPECIFICATION_ELEMENT_ID = "specificationElementId";

```

Comment:
Why do you opt for a static ID of the specification that is fixed in the code without also exposing a constructor where the ID could be changed? Is the plan to hide the ID of a specification from the user? If so, why?

Reply:
Sorry about, the thing with the comments. I will fix it.

About the specification ID:
In the current PR, the ID is a public static variable, so it is not about hiding it. Now that I think of it, I don't see any real reason to have a fixed spec ID. On the other hand, I also cannot think of a situation where giving the specification a certain ID would be useful in any way (with it being unique for each exploration anyway). 

The only situation where the ID may matter are potential name collisions where you could end up overwriting the spec in the Elements map. Having it as a constant, publically visible variable may reduce the risk of ID collisions, since there would be only one constant name to avoid when naming Elements.

All in all, I don't have a strong opinion into either direction. What do you think?

Reply:
Currently, Element comes with a constructor Element(String id) such that it is in the hand of the user to possibly handle (or be aware of) id conflicts. On the other hand, this also enables to use IDs for the sake of input/output handling, GUIs etc etc. I think a reasonable solution would be to use a generic "specification" as ID if none is given or also expose the constructor with String id in case the user wants to set a specification name/id as can be done for all Elements.